lib/feed/bloc/feed_bloc.dart

```diff
@@ -283,7 +283,33 @@ class FeedBloc extends Bloc<FeedEvent, FeedState> {
           return emit(state.copyWith(status: FeedStatus.failure));
         }
       case PostAction.report:
-      // TODO: Handle this case.
+        // Optimistically report the post
+        int existingPostViewMediaIndex = state.postViewMedias.indexWhere((PostViewMedia postViewMedia) => postViewMedia.postView.post.id == event.postId);
+
+        PostViewMedia postViewMedia = state.postViewMedias[existingPostViewMediaIndex];
+        PostView originalPostView = postViewMedia.postView;
+
+        try {
+          PostView updatedPostView = optimisticallyReportPost(postViewMedia, true);
```

Comment:
If `optimisticallyReportPost` simply marks the post as read, we should use `optimisticallyReadPost`. I would also add a comment here that indicates this is the action we want to take (otherwise, someone might mistake that as an error)